### ISSUE TITLE
PR creation failing from Org fork with  -  {"resource":"PullRequest","field":"head","code":"invalid"}

### ISSUE THREAD

**Issue by FoSix** [peter-evans/create-pull-request#2043]
I know it's similar issues were already posted, I've read them, didn't find a solution, so bare with me for a sec 😄 

### TL;DR

The problem goes down to this error:

```
Error: Validation Failed: {"resource":"PullRequest","field":"head","code":"invalid"}
```

being thrown when creating a PR from a fork which belongs to an organisation, the same organisation to which the base repo belongs to.

### SETUP

We have to repos under an organisation, both public, let's call them:

- `base` - a repo to which we would like introduce changes, we have read only access, hence changes have to be merged by a PR from a fork
- `CI` - hosting a workflow, triggered on `pull_request`

### WORKFLOW

The workflow is simple, when a PR is created in the CI repo:

- checkout the base repo
- introduce some changes
- run `create-pull-request` action to create a PR using a fork.

When the fork is a *personal* fork (not under organisation) everything works. The problem pops up when the fork is *inside* the same organisation as the base repo. 

PAT used for this is the classic type with `public_repo` access. I did try with all possible accesses but it does not seem to be an access problem. The account for which PAT was generated is a regular organisation member with owner access to CI repo and an owner of both forks. 

---

Below you can find links to test repos I used to reproduce the problem:

- [CI repo](https://github.com/fosixorg/ci)
- [base repo](https://github.com/fosixorg/base)
- [org fork](https://github.com/fosixorg/fork)
- [personal fork](https://github.com/fosix-bot/fork)
- [workflow](https://github.com/fosixorg/ci/blob/main/.github/workflows/ci.yml)
- [example run](https://github.com/fosixorg/ci/actions/runs/5190060961)

I did already spent a lot of time on this, so any help (even pointing out what I'm doing wrong) would be highly appreciated.

**Comment by peter-evans** [comment]
@FoSix Thankyou for your very detailed report of the issue. I notice that it has created the [pull request branch](https://github.com/fosixorg/base/compare/main...fosixorg:fork:create-pull-request/patch) successfully, but the PR creation failed. So it doesn't appear to be an issue related to token permissions.

My guess is that the action is struggling with the fact that the parent repo and fork are in the same org. That is not really normal since forks are usually made into a different org or user account. I'll investigate to see if there is something I can do to handle this case.

**Comment by FoSix** [comment]
I did some testing with with GH's REST API and `octokit/rest.js` and apparently it is possible to create a PR from a fork in the same org not only from UI (GH CLI however does not work). I'll skip the REST part, but for JS (assuming the same test repos):

```js
octokit.rest.pulls.create({
  owner: "fosixorg",
  repo: "base",
  head: "fosixorg:create-pull-request/patch",
  head_repo: "fosixorg/fork",
  base: "main",
  title: "script base PR",
  body: "created with GH script",
})
```

Proven to work [here](https://github.com/fosixorg/ci/actions/runs/5244144683/jobs/9469741435#step:2:2).

What does the trick is the `head_repo` parameter. And this is something that I think is [missing in the action's code](https://github.com/peter-evans/create-pull-request/blob/main/src/github-helper.ts#L52-L59). 

I'm not a JS developer so (unfortunately) I cannot propose a PR with code changes...

**Comment by peter-evans** [comment]
@FoSix Great work testing and finding that parameter! I've created a branch with the change to add `head_repo` in a PR [here](https://github.com/peter-evans/create-pull-request/pull/2044). It seems to be passing all of my tests. Would you be willing to test this version of the action and let me know if it solves your issue?

You can test the PR version of the action like this:
```yml
      - uses: peter-evans/create-pull-request@head-repo
```

If it works fine I'll merge it in and create a new release.

**Comment by FoSix** [comment]
> Would you be willing to test this version of the action and let me know if it solves your issue?

Sure!
